### PR TITLE
misc: reorganize the configuration file of nydusd

### LINKED ISSUE #1180: Nydusd can not start with the example config cause of missing some default params
Failed to run nydusd with the following example config on [nydusd.md](https://github.com/dragonflyoss/image-service/blob/master/docs/nydusd.md#run-with-fuse)
Error: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }

```
# Prepare nydusd configuration
cat /path/to/config-localfs.json
{
  "device": {
    "backend": {
      "type": "localfs",
      "config": {
        "dir": "/path/to/blobs",
      }
    }
  }
}
```

My test case:

```
# ./nydus-static/nydusd --config config-localfs.json --mountpoint nydus-mnt/ --bootstrap nydus-test/nodejs12-128.meta --log-level info
[2023-03-29 16:31:52.530243 +08:00] INFO Program Version: v2.2.0, Git Commit: "a38f6b8d6257af90d59880265335dd55fab07668", Build Time: "2023-03-01T10:05:57.267573846Z", Profile: "release", Rustc Version: "rustc 1.66.1 (90743e729 2023-01-10)"
[2023-03-29 16:31:52.530272 +08:00] INFO Set rlimit-nofile to 1000000, maximum 1000000
[2023-03-29 16:31:52.530427 +08:00] ERROR [/src/error.rs:22] Error:
	"failed to parse configuration information"
	at api/src/config.rs:232
	note: enable `RUST_BACKTRACE=1` env to display a backtrace
[2023-03-29 16:31:52.530441 +08:00] ERROR [/src/error.rs:22] Error:
	Rafs(LoadConfig(Os { code: 22, kind: InvalidInput, message: "Invalid argument" }))
	at service/src/lib.rs:121
	note: enable `RUST_BACKTRACE=1` env to display a backtrace
[2023-03-29 16:31:52.530461 +08:00] ERROR [src/bin/nydusd/main.rs:530] Failed in starting daemon: Invalid argument (os error 22)
Error: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }

# cat config-localfs.json
{
  "device": {
    "backend": {
{
      "type": "localfs",
      "config": {
        "dir": "/root/nydus-test/",
      }
    }
  }
}

# ls /root/nydus-test/
nodejs12-128.blob  nodejs12-128.meta  nodejs12-256.blob  nodejs12-256.meta
```

The reason is that some default params are missing. Using the following config.json works well.
```
cat config-localfs.json
{
  "device": {
    "backend": {
      "type": "localfs",
      "config": {
        "dir": "/root/nydus-test"
      }
    },
    "cache": {
      "type": "blobcache",
      "config": {
        "work_dir": "/cache"
      }
    }
  },
  "mode": "direct",
  "digest_validate": false,
  "iostats_files": true,
  "enable_xattr": true,
  "fs_prefetch": {
    "enable": true,
    "threads_count": 4
  }
}
```

### PR DESCRIPTION
**Pull request by adam3q**
Fix: #1180
1. Move configuration files from docs/samples to misc/configs
2. Fix incomplete configuration in docs/nydusd.md
3. Update outdated nydusd-config.json from nydus-snapshotter repo

### CODE REVIEW

File: docs/nydusd.md [hunk 1]
```diff
@@ -8,27 +8,42 @@ Get `nydusd` binary from [release](https://github.com/dragonflyoss/image-service
...
+    "cache": {
+      "type": "blobcache"
     }
+  },
+  "mode": "direct",
+  "digest_validate": false,
+  "iostats_files": false,
+  "enable_xattr": true,
+  "fs_prefetch": {
+    "enable": true,
```
[the remark this answers is not in the document]
Comment (adamqqqplay) [reply docs/nydusd.md:31]: @bergwolf I copied the config file from [here](https://github.com/containerd/nydus-snapshotter/blob/main/misc/snapshotter/nydusd-config-localfs.json), are these options unnecessary?

[the remark this answers is not in the document]
Comment (adamqqqplay) [reply docs/nydusd.md]: @bergwolf I removed this part from config.

File: docs/nydusd.md [hunk 2]
```diff
@@ -8,27 +8,41 @@ Get `nydusd` binary from [release](https://github.com/dragonflyoss/image-service
...
 # Prepare nydusd configuration
-cat /path/to/config-localfs.json
+sudo tee /etc/nydus/nydusd-config.localfs.json > /dev/null << EOF
 {
   "device": {
     "backend": {
       "type": "localfs",
       "config": {
-        "dir": "/path/to/blobs",
+        "dir": "/tmp/blobs/"
```
Comment (jiangliu) [inline docs/nydusd.md:19]: how about using `/var/lib/nydus/blobs` here?

File: misc/configs/nydusd-config.json [hunk 3]
```diff
@@ -3,25 +3,23 @@
...
-        "scheme": "http",
         "timeout": 5,
         "connect_timeout": 5,
-        "retry_limit": 0
+        "retry_limit": 2
       }
     },
     "cache": {
-      "type": "blobcache",
-      "config": {
```
Comment (jiangliu) [inline misc/configs/nydusd-config.json:14]: why the `config` field get removed?

Comment (adamqqqplay) [reply misc/configs/nydusd-config.json:14]: @jiangliu After this [commit](https://github.com/containerd/nydus-snapshotter/commit/c5e1c57766aa4ac96fe80f7ec2c09223af72d845) in https://github.com/containerd/nydus-snapshotter/pull/229, the `work_dir` field is unnecessary.

File: docs/nydusd.md (hunk 2 above)
Comment (adamqqqplay) [reply docs/nydusd.md:19]: Looks good. I fixed it.

File: misc/configs/nydusd-config.json (hunk 3 above)
Comment (adamqqqplay) [reply misc/configs/nydusd-config.json:14]: @jiangliu I misunderstood before, this field is still required when there is no snapshotter, it has been fixed now.

**Review by bergwolf — APPROVED** [review]

**Review by jiangliu — APPROVED** [review]

**Pull request merged**